jolt-core/src/jolt/vm/mod.rs

```diff
@@ -186,8 +188,114 @@ pub trait Jolt<F: PrimeField, G: CurveGroup<ScalarField = F>, const C: usize, co
         )
     }
 
-    fn prove_r1cs() {
-        unimplemented!("todo")
+    fn prove_r1cs(
+        instructions: Vec<Self::InstructionSet>,
+        mut bytecode_rows: Vec<ELFRow>,
+        mut trace: Vec<ELFRow>,
+        bytecode: Vec<ELFInstruction>,
+        memory_trace: Vec<MemoryOp>,
+        circuit_flags: Vec<F>,
+        transcript: &mut Transcript,
+        random_tape: &mut RandomTape<G>,
+    ) {
+        let N_SKIP = 3;  // the instructions at the beginning of the trace to be skipped 
+        let N_FLAGS = 17;
+        let TRACE_LEN = trace.len()-N_SKIP;
+
+        let log_M = log2(M) as usize;
+
+        let instructions = &instructions[N_SKIP..];
+        let circuit_flags = &circuit_flags[N_FLAGS * N_SKIP..];
+
+        let [mut prog_a_rw, mut prog_v_rw, prog_t_reads] = 
+            BytecodePolynomials::<F, G>::r1cs_polys_from_bytecode(bytecode_rows, trace, N_SKIP);
+        prog_v_rw.extend(circuit_flags
+            .chunks(N_FLAGS)
+            .map(|x| {
+                let mut packed = F::ZERO;
```

Comment:
Nit: Use `F::zero()` to be consistent with rest of codebase.

Reply:
Changed it here but wasn't possible in parts of `r1cs/snark.rs` though. Might be cause that's ff::PrimeField.